packages/vm/lib/state/interface.ts

```diff
@@ -32,4 +32,8 @@ export interface StateManager {
   accountExists(address: Address): Promise<boolean>
   cleanupTouchedAccounts(): Promise<void>
   clearOriginalStorageCache(): void
+  addWarmAddress(address: Buffer): void
+  isWarmAddress(address: Buffer): boolean
+  addWarmStorage(address: Buffer, slot: Buffer): void
+  isWarmStorage(address: Buffer, slot: Buffer): boolean
```

Comment:
These methods unfortunately can't be added here yet to be mandatory, since this would be a breaking change.

Just sitting here realizing that this gets - again - a bit trickier from the compatibility angle. Suggestion: we add these here as optional and then make it very clear in the code docs /release notes, that - if people want to run EIP2929 (and likely also 2930) functionality AND have their custom state manager implementation - they need to implement these methods.

Ug. Likely also always (?) a good advice for people to do their custom implementations by inheriting from the default state manager. Then this has a higher chance that a custom state manager will continue to work on such kind of updates.

Reply:
Yep you are right. I only fear that if I remove this, then TypeScript is going to complain about possible non existent methods...

Reply:
Yeah, that's what we have to live with and what the `!` operator is for. Unfortunately no way around that we just can't do such an interface change along the road without a major VM release. 😕

Reply:
Sadly we cannot use the ! on methods. (I tried). Now we have to resort to explicit casting or casting the state manager as any =(